lua/nvim-tree/utils.lua

```diff
@@ -59,7 +59,15 @@ end
 ---@param relative_to string
 ---@return string
 function M.path_relative(path, relative_to)
-  local p, _ = path:gsub("^" .. M.path_to_matching_str(M.path_add_trailing(relative_to)), "")
+  local _, r = path:find(M.path_add_trailing(relative_to), 1, true)
+  local p = path
+  if r and r < p:len() then
```

Comment:
Looking good... except when there is a trailing slash.

See test case attached.
[2005-test.lua.gz](https://github.com/nvim-tree/nvim-tree.lua/files/10776248/2005-test.lua.gz)

Reply:
Thanks for the test cases. Just pushed the fix.  previously version if relative == path, I kept the path but to match the original gsub version, the code now just returns ""

Reply:
Fantastic, thank you.

Reply:
no problem, thanks for the excellent plugin!